src/data/queries/mediaImage.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  QueryData,
+  QueryFormatter,
+  QueryOpts,
+  QueryParams,
+} from 'next-drupal-query'
+import { queries } from '.'
+import { MediaImage } from '@/types/dataTypes/drupal/media'
+import { drupalClient } from '@/lib/utils/drupalClient'
+import { MediaImageType } from '@/types/index'
+
+// Define query params for queryData.
+export const params: QueryParams<null> = () => {
+  return queries.getParams().addPageLimit(12).addInclude(['image'])
+}
+// Define the option types for the data loader.
+type DataOpts = QueryOpts<null>
+
+// Implement the data loader.
+export const data: QueryData<DataOpts, MediaImage[]> = async (): Promise<
+  MediaImage[]
+> => {
+  const entities = await drupalClient.getResourceCollection<MediaImage[]>(
+    'media--image',
+    {
+      params: params().getQueryObject(),
+    }
+  )
+  return entities
```

Comment:
are there fields with multiple images that require this to be an array vs single media object? a little unclear to me

Reply:
This was confusing to me as well yes there are both types and I had some issues mapping though the data in the query service so I added the logic to mediaImage component directly. Let me know if you see a better path

Reply:
Found a better way to handle this